testing/hierarchies.py

```diff
@@ -190,6 +194,30 @@ def __str__(self):
         return "{0}: {1}".format(self.kind, self.location)
 
 
+class page(node):  # noqa: N801
```

Comment:
I had to add this class because doxygen's nested pages couldn't be represented in the test as files, since files can't be defined in other files. This is for the alt_town_rock.hpp tests

Reply:
OK, once things are actually finished here I think I want to splice out the page changes to a separate PR.  Did you take any of this from #157?

Reply:
Yes, I took it from #157 and added some debugging by keeping track of the location, but it was really only to get the tests to pass :)
Can definitely be spliced out to the other PR.

Reply:
Ok, that's no problem. Are you using doxygen 1.9.2?

We're in a fragile place with CI, more recent doxygen breaks a lot of things :/